lib/ReactViews/Analytics/ParameterEditor.jsx

```diff
@@ -37,124 +208,15 @@ const ParameterEditor = React.createClass({
-                        />
-                    </div>);
-            case 'region':
-                return (
-                    <div>
-                        {this.renderLabel()}
-                        <RegionParameterEditor
-                            previewed={this.props.previewed}
-                            viewState={this.props.viewState}
-                            parameter={this.props.parameter}
-                        />
-                    </div>);
-            case 'regionType': {
-                const that = this;
-                const regionParam = this.props.previewed.parameters.find(function(param) {
-                    return (defined(param.regionTypeParameter) &&
-                            param.regionTypeParameter === that.props.parameter);
-                });
-                return (
-                    <div>
-                        <If condition={regionParam === undefined}>
-                            {this.renderLabel()}
-                            <RegionTypeParameterEditor
-                                previewed={this.props.previewed}
-                                parameter={this.props.parameter}
-                            />
-                        </If>
-                        <If condition={!this.props.parameter.showInUi}>
-                            <div className="Placeholder for regionType"/>
-                        </If>
-                    </div>);
+        for (let i = 0; i < this.parameterTypeConverters.length; ++i) {
+            const converter = this.parameterTypeConverters[i];
+            const editor = converter.parameterTypeToDiv(this.props.parameter.type, this);
+            if (defined(editor)) {
+                return editor;
             }
-            case 'regionData':
-                return (
-                    <div>
-                        {this.renderLabel()}
-                        <RegionDataParameterEditor
-                            previewed={this.props.previewed}
-                            parameter={this.props.parameter}
-                        />
-                    </div>);
-            case 'boolean':
-                return (
-                    <div>
-                        {this.renderLabel()}
-                        <BooleanParameterEditor
-                            previewed={this.props.previewed}
-                            parameter={this.props.parameter}
-                        />
-                    </div>);
-            default:
-                return (
-                    <div>
-                        {this.renderLabel()}
-                        <GenericParameterEditor
-                            previewed={this.props.previewed}
-                            parameter={this.props.parameter}
-                        />
-                    </div>);
         }
+        const genericEditor = this.parameterTypeConverters.find(function(item) { return item.id === 'generic'; });
```

Comment:
`find` isn't available on Internet Explorer.  There's a `findIndex` function in TerriaJS that you can use instead, or you can do:
`this.parameterTypeConverters.filter(function(item) { return item.id === 'generic'; })[0];`